typescript/sdk/src/ism/EvmIsmReader.ts

```diff
@@ -66,6 +67,7 @@ export class EvmIsmReader extends HyperlaneReader implements IsmReader {
     protected readonly concurrency: number = multiProvider.tryGetRpcConcurrency(
       chain,
     ) ?? DEFAULT_CONTRACT_READ_CONCURRENCY,
+    protected readonly messageContext?: DispatchedMessage,
   ) {
```

Comment:
I think that it would more reusable to add the `messageContext` to the `deriveRoutingConfig` method signature instead of tying the class instance to a specific message. I don't have a use case in my mind right now but this way instead of having to create a new instance of the class we would just need to call `deriveRoutingConfig` with a different message context and be able to reuse the same instance

Reply:
the `messageContext` needs to be propagated across all of the recursive function calls so this was the "least invasive" way for me to make this change
would you prefer it to be optional on every function?